### DISCUSSION TITLE
AST mapping detail

### DISCUSSION THREAD

**Discussion by micttyl** [carbon-language/carbon-lang#2124] [Getting started] [answered]
Is there any rough layout of node-to-node AST mapping between Clang and Carbon to get some common idea?

By the way, I see Carbon separated Statement and Expression. I heard this was a desired improvement of Clang AST.

**Comment by jonmeow** [comment, marked as answer]
I don't think anyone is maintaining something like this. It's not something I could quickly write up either, I just don't know the Clang AST that well.

Note, you're probably looking at the explorer's AST -- the toolchain's direction is more significantly different. It's a deliberate shift away from the AST focus.

**Reply by micttyl** [reply]
Thank you. I'll go take a look.
